### PR TITLE
feat: improve card semantics (resolves #190)

### LINKED ISSUE #190: Review design and implementation of cards and info cards for improved accessibility
**Describe the bug**
The cards and info-cards are accessible, but not the best accessibility experience. For example, in many cases when a card is focused or read aloud, non-critical information is presented first (like the category or a close button), then the title of the card.

Also each hard has a lot of semantics, some of which may not be critical for a non-visual user. There is an opportunity to simplify and tighten up the semantic information.

**To Reproduce**
1. Launch a screen reader
2. Have the screen reader read through resource cards and info-cards.
3. Notice the complexity of cards (headers, lists, etc.)
4. Notice the order in which non-critical information is presented to the user first.

**Expected behavior**
The feedback to the user can be simplified to make it easier to cognitively understand the content. Critical information to the user should be presented first, then non-critical information.

### LINKED ISSUE #193: Atom: favorites indicator
## Is your feature request related to a problem? Please describe.

We need to represent whether a resource is in a user's favorites list or not on both the card and single resource view.

## Describe the solution you'd like

The heart icon plus accessible text can be used to reflect this independently from the "Add to / Remove from favorites" button.

## Describe alternatives you've considered

Not applicable.

## Additional context

Not applicable.

### PR DESCRIPTION
**Pull request by greatislander**
* [x] I've read the [guidelines for Contributing to the Platform Co-op Toolkit](https://github.com/platform-coop-toolkit/.github/blob/master/CONTRIBUTING.md)
* [x] This isn't a duplicate of an existing pull request

## Description

Applies changes from #190 to all cards.

TODO: Add icon to byline.

## Steps to test

Review all card and archive components: https://deploy-preview-209--pinecone.netlify.com

## Additional information

Not applicable.

## Related issues

- Resolves #190.
- Resolves #193.

### CODE REVIEW

**Comment by cherylhjli** [comment]
the page format looks great! the location looks like its hanging a bit lower than the other ones in the individual card component?

**Comment by greatislander** [comment]
Good catch @cherylhjli!

**Comment by greatislander** [comment]
@cherylhjli This is done! I just need the icon for the byline.

**Comment by jhung** [comment]
I think the "+2 more" should be part of the Topics list. Right now "+2 more" appears out of context. For example, NVDA would say:

> "... List item topic commons. List item topic policy. Out of list. Plus two more"

It may make more sense if the feedback says:

> "... List item topic commons. List item topic policy. Plus two more topics. Out of list."

File: src/components/02-molecules/01-card/card--resource.njk [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+				<h3 class="card__title" >
+					<a href="{{ href }}">{{ title | safe }}</a>
+				</h3>
+				{% if byline %}
+					<p class="card__byline">By {{ byline | safe }}</p>
+				{% endif %}
+			</header>
+			<aside class="card__aside">
+				<div class="card__meta">
+					<span class="card__format">{% render '@svg', {svg:formatIcon}, true %}{{ format }}</span>
```
Comment (jhung) [inline src/components/02-molecules/01-card/card--resource.njk]: Add screen reader label: "Resource format:".

**Comment by greatislander** [comment]
@jhung I think I've addressed all the review comments (including adding screen reader-accessible labels to the other cards).

**Comment by jhung** [comment]
One last thing, the mid-dot does not cause the screen reader to pause. This makes "resource format article (mid-dot) publisher" come out as a single uninterrupted string of words which is confusing.

I tried to experiment with this by adding `aria-label="."` to `<span class="card__sep">` but NVDA just ignores it.

Could a regular period be used instead of a mid-dot and offset from the baseline to visually appear like a mid-dot? (Personally, even an unstyled period looks fine, but maybe @cherylhjli has thoughts?)

A hacky way to do this would be to add a non-visual span and put a period.

**Comment by cherylhjli** [comment]
Could we move the baseline of the period up, so the period resembles a mid-dot?

**Comment by greatislander** [comment]
@cherylhjli @jhung I just tested with VoiceOver/Safari and realized that VoiceOVer reads 'middle dot' without pausing… so definitely an alternate approach is needed. I'm going to try the period trick.

**Comment by greatislander** [comment]
Using a period works great in Safari with Voiceover. One observation: this looks weird, but maybe it's okay?

Screen shot of VoiceOver modal which reads: 'language: English . publication date: 2016'.

**Comment by jhung** [comment]
> 
> 
> Using a period works great in Safari with Voiceover. One observation: this looks weird, but maybe it's okay?
> 
> Screen shot of VoiceOver modal which reads: &apos;language: English . publication date: 2016&apos;.

Is there any way to offset the period vertically so it appears in the middle of the line (like a mid-dot)?

**Comment by greatislander** [comment]
Yes, I can do that with CSS (set `vertical-align: super`). However that doesn't affect the VoiceOver modal.

**Comment by greatislander** [comment]
See latest changes from 2120c96.

File: src/components/02-molecules/01-card/card--resource.njk [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+				</h3>
+				{% if byline %}
+					<p class="card__byline">By {{ byline | safe }}</p>
+				{% endif %}
+			</header>
+			<aside class="card__aside">
+				<div class="card__meta">
+					<span class="card__format"><span class="screen-reader-text">resource format: </span>{% render '@svg', {svg:formatIcon}, true %}{{ format }}</span>
+					{% if publisher %}
+					{% render '@separator' %}
```
Comment (jhung) [inline src/components/02-molecules/01-card/card--resource.njk]: It seems that the presence of white space around the separator is forcing NVDA to say "dot" for the separator. If you remove the white space before the period, NVDA will pause.

To test this, I moved line 18 and appended it to line 16 without any printing space between spans.

Comment (greatislander) [reply src/components/02-molecules/01-card/card--resource.njk]: Okay, will do.

**Pull request merged**